Lib/test/test_complex.py

```diff
@@ -21,6 +21,27 @@
     (1, 0+0j),
 )
 
+class MockIndex:
```

Comment:
Naming nitpick: the `Mock` prefix on these classes doesn't feel appropriate: nothing's being mocked here. I'd probably go with something like `ImplementsIndex` or `HasIndex`.

Reply:
I once added common `FakePath` class which implements the `__path__` method, and it is now used in many tests for the path protocol. I proposed to add similar FakeIndex, FakeFloat, etc, but Raymond rejected this idea. Here I use the `Mock` prefix which looks more neutral and test specific than `Fake`. Different tests currently use similar classes with different names. The most common prefix is perhaps `My`. Prefix `Supports` cannot be used because it can conflict with `typing.SupportIndex`, etc. Suffix `Like` can conflict with `os.PathLike`.

I see that several tests use the `With` prefix: `WithStr`, `WithRepr`, `WithIterAnext`, etc. What do you think about `WithIndex`?

Reply:
`WithIndex` works. I'm also happy for this to be left as-is - it's not a strong objection, just a niggling feeling that the name isn't quite right.